### DISCUSSION TITLE
NLS parameter setup in oracledb environment level

### DISCUSSION THREAD

**Discussion by avmcohen** [oracle/python-oracledb#256] [Q&A] [answered]
Hello
There is an option to set **NLS_DATE_FORMAT** in Unix/OS level - that effect any sqlplus connection in this session.
Connecting by oracledb with these setup - not effecting the DB initial setup.
See below test case - which I set the NLS_DATE_FORMAT to YYYYMMDD - but still show in oracledb connection as DD-MON-RR.
Any advice ?..

_**Test Case:**
<UNIX> export NLS_DATE_FORMAT='**YYYYMMDD**' 
<UNIX> export NLS_LANG=American_America.AL32UTF8
<UNIX>python3
Python 3.9.5 (default, May 13 2021, 09:57:24)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39.0.3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import oracledb as odb
>>> odb.init_oracle_client()
>>> dbs = odb.connect("<USER>/***@<DB>")
>>> dbc = dbs.cursor()
>>> dbc.execute("SELECT * from nls_database_parameters where PARAMETER = 'NLS_DATE_FORMAT'").fetchall()
[('NLS_DATE_FORMAT', '**DD-MON-RR'**)]_

**Comment by cjbj** [comment, marked as answer]
I presume you are using Thin mode.  If so, the options are to use Thick mode, or read the doc on [Thin Mode Locale-aware Number and Date Conversions](https://python-oracledb.readthedocs.io/en/latest/user_guide/globalization.html#thin-mode-locale-aware-number-and-date-conversions).
